runtime/vm/segment.c

```diff
@@ -427,7 +438,17 @@ static J9MemorySegment * allocateVirtualMemorySegmentInListInternal(J9JavaVM *ja
 				omrthread_jit_write_protect_enable();
 			}
 			segment->baseAddress = allocatedBase;
-			segment->heapBase = allocatedBase;
+#if defined(J9VM_OPT_CRIU_SUPPORT) && defined(LINUX)
+			if (J9_ARE_ANY_BITS_SET(segment->type, MEMORY_TYPE_RAM_CLASS | MEMORY_TYPE_ROM_CLASS)
+				&& J9_ARE_ALL_BITS_SET(javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_CRIU_DISCLAIM_CLASSES)
+			) {
+				/* Adjust address to achieve 4K alignment. */
+				segment->heapBase = (uint8_t *)ROUND_TO(J9SEGMENT_ALIGNMENT, (UDATA)allocatedBase);
```

Comment:
does this work if its `allocatedBase` is less than 4k?

Reply:
also, if you request 6k and its unaligned, the you'll need to bump it up to the nearest 4k boundary. Will you still return enough more to fulfill the requested size?

Reply:
in `allocateMemoryForSegment`, more memory (=4K) is being requested through `j9mem_allocate_memory` for the alignment. over here, we just bump the address to the nearest 4K boundary.